burdigalax/client/inventory.lua

```diff
@@ -0,0 +1,223 @@
+local inventoryUI
+
+function OnPackageStartInventory()
+	local width, height = GetScreenSize()
+	inventoryUI = CreateWebUI(0, 0, 0, 0, 10, 60)
+    LoadWebFile(inventoryUI, "http://asset/onsetrp/burdigalax/client/gui/inventory.html")
+	SetWebAlignment(inventoryUI, 0.0, 0.0)
+	SetWebAnchors(inventoryUI, 0.0, 0.0, 1.0, 1.0)
+	SetWebVisibility(inventoryUI, WEB_HIDDEN)
+end
+AddEvent("OnPackageStart", OnPackageStartInventory)
+
+function OnPackageStopInventory()
+	DestroyWebUI(inventoryUI)
+end
+AddEvent("OnPackageStop", OnPackageStopInventory)
+
+-- CLOSE
+
+function CloseUIInventory(context)
+    personalMenuIsOpen = 0
+    if context == '"transfer"' then
+        ExecuteWebJS(inventoryUI, "ResetSelectedNearbyInventory()")
+    else
+        ShowMouseCursor(false)
+        SetInputMode(INPUT_GAME)
+        Delay(100, function()
+            SetWebVisibility(inventoryUI, WEB_HIDDEN)
+        end)
+    end
+end
+AddEvent("BURDIGALAX_inventory_onClose", CloseUIInventory)
+
+-- INIT
+
+function OpenUIInventory(key, items, playerInventory, playerName, playerId, playersList, maxSlots)
+    personalMenuIsOpen = 1
+    ExecuteWebJS(inventoryUI, "BURDIGALAX_inventory.setConfig("..json_encode(BuildInventoryJson(items, playerInventory, playerName, playerId, playersList, maxSlots))..");")
+    ShowMouseCursor(true)
+    SetInputMode(INPUT_GAMEANDUI)
+    SetWebVisibility(inventoryUI, WEB_VISIBLE)
+end
+
+-- UPDATE
+
+AddRemoteEvent("UpdateUIInventory", function(player, item, quantity)
+    ExecuteWebJS(inventoryUI, "BURDIGALAX_inventory.updateItemsInventories("..player..", [{ id: '"..item.."', quantity: "..quantity.." }]);")
+end)
+
+function onEquipItemInventory(event)
+    local data = json_decode(event)
+    ExecuteWebJS(inventoryUI, 'setEquipItemInventory('..data.idInventory..','..data.idItem..','..tostring(data.isEquipped)..');')
+end
+AddEvent('BURDIGALAX_inventory_onEquip', onEquipItemInventory)
+
+-- TRANSFERT
+
+function onTransferItems(event)
+    local data = json_decode(event)
+    ExecuteWebJS(inventoryUI, "setUpdateItemsInventories("..data.destinationInventoryId..","..data.idItem..","..data.newQuantityDestination..");")
+    ExecuteWebJS(inventoryUI, "setUpdateItemsInventories("..data.originInventoryId..","..data.idItem..","..data.newQuantityOrigin..");")
+end
+AddEvent('BURDIGALAX_inventory_onTransfer', onTransferItems)
+
+function BuildInventoryJson(items, playerInventory, playerName, playerId, playersList, maxSlots)
+    local json = {
+        config = {
+            hasEquipableCategory = true
+        },
+        mainInventoryId = playerId,
+        items = InventoryItems(items),
```

Comment:
rename methods getInventoryItems, getInventoryEffects, getInventoryCategories, getInventories for more explicit code.